enumerable.rb

```diff
@@ -0,0 +1,142 @@
+# rubocop:disable Metrics/ModuleLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Style/CaseEquality
+module Enumerable
+  def my_each
+    if block_given?
+      length.times do |el|
+        yield self[el]
+      end
+      self
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      length.times do |indx|
+        yield self[indx], indx
+      end
+      self
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    result = []
+    if block_given?
+      my_each do |el|
+        result << el if yield el
+      end
+      result
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(arg = nil)
+    my_each do |value|
+      if block_given?
+        return false unless yield value
+      elsif arg.class == Class
+        return false unless arg === value
+      elsif arg.class == Regexp
+        return false unless arg =~ arg
+      else
+        return false unless value == arg
+      end
+    end
+    true
+  end
```

Comment:
- [ ] my_all? - When no block or argument is given returns true when none of the collection members are false or nil

`[nil, true, 99].my_all?` should return false
`[].my_all?` should return true

- [ ] my_all? - When a Regex is passed as an argument returns true if all of the collection matches the Regex

`%w[ant bear cat].all?(/t/)`  should return false
`%w[ant tiger cat].all?(/t/)`  should return true